### ISSUE TITLE
Handling Power Automate Fetch XML Query Limitation

### ISSUE THREAD

**Issue by charlie-iu** [charlie-iu/blog#28]
# Handling Power Automate Fetch XML Query Limitation

Power Automate is a powerful tool for automating workflows, and when it comes to querying data from Dynamics 365 using Fetch XML, it's essential to be aware of a potential limitation. The default Fetch XML generated by the XRM Tool often includes a `top=50` clause, limiting the query results to 50 records. Failing to remove this clause can lead to issues when the returned data exceeds this limit.

## The Fetch XML Limitation

When using the XRM Tool to generate Fetch XML queries in Power Automate, the resulting XML may contain the following snippet:

```xml
<fetch version="1.0" output-format="xml-platform" mapping="logical" distinct="false">
  <entity name="your_entity">
    <!-- other conditions and attributes -->
    <fetch count="50" >
      <!-- your fetch attributes -->
    </fetch>
  </entity>
</fetch>
```
The Problem
The count="50" attribute in the <fetch> element imposes a restriction on the query result set, limiting it to 50 records. This can be problematic if your dataset exceeds this threshold, as only a partial result set will be returned.

Solution: Removing the top=50 Clause
To ensure that your Fetch XML query retrieves the complete dataset, it's crucial to remove the count="50" attribute. Modify the Fetch XML as follows:
```xml<fetch version="1.0" output-format="xml-platform" mapping="logical" distinct="false">
  <entity name="your_entity">
    <!-- other conditions and attributes -->
    <fetch >
      <!-- your fetch attributes -->
    </fetch>
  </entity>
</fetch>
```
By eliminating the count="50" attribute, you allow the query to retrieve all records that match the specified conditions without the imposed limit.

Best Practices
When working with large datasets, it's advisable to optimize your queries and consider implementing pagination or incremental loading strategies to manage the data effectively. This ensures efficient processing and prevents potential timeouts or performance issues.

In conclusion, being mindful of the default top=50 limitation in Fetch XML queries generated by the XRM Tool in Power Automate is crucial for avoiding incomplete result sets. By removing this clause, you ensure the accurate retrieval of data, especially when dealing with datasets exceeding 50 records.
